File: new/cpp/121_best_time_to_buy_and_sell_stock_brute_force.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        // idea: brute force
        // calculate profits for each buy and sell day where buy day < sell day
        int profits = 0;
        for (int buy_day = 0; buy_day < prices.size(); buy_day++) {
            int buy_price = prices[buy_day];
            for (int sell_day = buy_day + 1; sell_day < prices.size(); sell_day++) {
                int sell_price = prices[sell_day];
                profits = max(profits, sell_price - buy_price);
            }
        }

        return profits;
    }
    // time: O(n^2)
    // space: O(1)
};

File: new/python/121_best_time_to_buy_and_sell_stock_brute_force.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # idea: brute force
        # track profits for each step
        # during each step of the iteration, find the difference between buying and selling of all possible combinations in the given
        # list following the rule (selling day > buying day)
        current_profit = 0
        for buy_date in range(len(prices)):
            buy = prices[buy_date]
            for sell_date in range(buy_date + 1, len(prices)):
                sell = prices[sell_date]
                current_profit = max(current_profit, sell - buy)

        return current_profit
        # time: O(n^2)
        # space: O(1)
